checks/branch_protected_test.go

```diff
@@ -32,12 +35,302 @@ func (l *log) Logf(s string, f ...interface{}) {
+					},
+					EnforceAdmins: &github.AdminEnforcement{
+						URL:     nil,
+						Enabled: false,
+					},
+					Restrictions: &github.BranchRestrictions{
+						Users: nil,
+						Teams: nil,
+						Apps:  nil,
+					},
+					RequireLinearHistory: &github.RequireLinearHistory{
+						Enabled: false,
+					},
+					AllowForcePushes: &github.AllowForcePushes{
+						Enabled: false,
+					},
+					AllowDeletions: &github.AllowDeletions{
+						Enabled: false,
+					},
+				},
+			},
+			want: checker.CheckResult{
+				Name:        CheckBranchProtection,
+				Pass:        false,
+				Details:     nil,
+				Confidence:  7,
+				ShouldRetry: false,
+				Error:       nil,
+			},
+		},
+		{
+			name:     "Both release and development are OK",
+			releases: []*string{&rel1},
+			protections: map[string]*github.Protection{
+				"main": {
+					RequiredStatusChecks: &github.RequiredStatusChecks{
+						Strict:   true,
+						Contexts: []string{"foo"},
+					},
+					RequiredPullRequestReviews: &github.PullRequestReviewsEnforcement{
+						DismissalRestrictions: &github.DismissalRestrictions{
+							Users: nil,
+							Teams: nil,
+						},
+						DismissStaleReviews:          true,
+						RequireCodeOwnerReviews:      true,
+						RequiredApprovingReviewCount: 1,
+					},
+					EnforceAdmins: &github.AdminEnforcement{
+						URL:     nil,
+						Enabled: true,
+					},
+					Restrictions: &github.BranchRestrictions{
+						Users: nil,
+						Teams: nil,
+						Apps:  nil,
+					},
+					RequireLinearHistory: &github.RequireLinearHistory{
+						Enabled: true,
+					},
+					AllowForcePushes: &github.AllowForcePushes{
+						Enabled: false,
+					},
+					AllowDeletions: &github.AllowDeletions{
+						Enabled: false,
+					},
+				},
+				"release/v.1": {
+					RequiredStatusChecks: &github.RequiredStatusChecks{
+						Strict:   true,
+						Contexts: []string{"foo"},
+					},
+					RequiredPullRequestReviews: &github.PullRequestReviewsEnforcement{
+						DismissalRestrictions: &github.DismissalRestrictions{
+							Users: nil,
+							Teams: nil,
+						},
+						DismissStaleReviews:          true,
+						RequireCodeOwnerReviews:      true,
+						RequiredApprovingReviewCount: 1,
+					},
+					EnforceAdmins: &github.AdminEnforcement{
+						URL:     nil,
+						Enabled: true,
+					},
+					Restrictions: &github.BranchRestrictions{
+						Users: nil,
+						Teams: nil,
+						Apps:  nil,
+					},
+					RequireLinearHistory: &github.RequireLinearHistory{
+						Enabled: true,
+					},
+					AllowForcePushes: &github.AllowForcePushes{
+						Enabled: false,
+					},
+					AllowDeletions: &github.AllowDeletions{
+						Enabled: false,
+					},
+				},
+			},
+			want: checker.CheckResult{
+				Name:        CheckBranchProtection,
+				Pass:        true,
+				Details:     nil,
+				Confidence:  10,
+				ShouldRetry: false,
+				Error:       nil,
+			},
+		},
+	}
+
+	get = func(context.Context, string, string) (*github.Repository,
```

Comment:
The `get`, `listReleases` and `getBranchProtection` fn vars are confusing. Instead, update `mockRepos` struct to hold the data passed in through the testcase. Something like:

```
type mockRepos struct {
  defaultBranch string
  releases []*github.RepositoryRelease
  protections map[string]*github.Protection
}
```

The interface fns can directly work on these internal vars. Then, your `mockRepo` initialization would simply look like:

```
mockrepo := mockRepo {
  defaultBranch: tt.defaultBranch,
  releases: tt.releases,
  protections: tt.protections,
}
```